uefi-test-runner/src/proto/console/serial.rs

```diff
@@ -35,6 +37,15 @@ pub fn test(bt: &BootServices) {
         assert_eq!(len, msg_len, "Serial port read timed-out!");
 
         assert_eq!(&output[..], &input[..msg_len]);
+
+        // Clean up after ourselves
+        serial.reset().expect("Could not reset the serial device");
+        let settable_bits = ControlBits::REQUEST_TO_SEND
```

Comment:
I wonder if it would be more convenient to the user if we performed this AND-ing in the function itself? This way there would be no way for the user to mistakenly use a forbidden bit.

Reply:
I initially went in that direction, then I figured out that I did not want to involuntarily hide user errors. After all, if the user tries to set a bit which is not settable, he ought to know.

If this were Ada, or another language with easy-to-use refinement typing, I would modify the signature of set_control_bits so that it only accepts as input a subset of the values of ControlBits. In Rust, this can be emulated with a wrapper type, but I'm not sure if that's worth the trouble (you end up with ugly calls like `set_control_bits(SettableControlBits::new(...))`).

Reply:
Alright, then it's up to the user of the API.

Reply:
OTOH, a straightforward interface change would be to provide a bitmask for the supported control bits in the serial protocol interface. Let me do that...